client/py/anvil/util.py

```diff
@@ -84,15 +84,19 @@ def __deepcopy__(self, memo):
 @serializable_type
 class WrappedList(list):
     def __init__(self, lst=[]):
-        for x in lst:
-            self.append(x)
+        if isinstance(lst, (WrappedObject, WrappedList)):
+            list.__init__(self, lst)
+        else:
+            list.__init__(self, map(_wrap, lst))
```

Comment:
In principle, further refactor for `_wrap` would clean this up a bit. This is the place to target:

```
        wl = WrappedList()
        for i in value:
            wl.append(i)
        return wl
```

If this is optimized, I don't think you would need the short-circuiting here -- `_wrap` has short-circuiting.

(One could argue that it's cleaner to refactor `_wrap` first, get the performance improvement there, then do a slightly cleaner refactor/performance improvement here.)